diffkemp/simpll/Result.h

```diff
@@ -106,19 +120,37 @@ struct NonFunctionDifference {
 
 /// Syntactic difference between objects that cannot be found in the original
 /// source files.
-/// Note: this can be either a macro difference or inline assembly difference.
+/// There are multiple kinds of the differences: a macro diff,
+/// inline assembly diff, function-macro diff or macro-function diff.
 struct SyntaxDifference : public NonFunctionDifference {
+    enum SyntaxKind {
+        MACRO,
+        ASSEMBLY,
+        FUNCTION_MACRO,
+        MACRO_FUNCTION,
+        UNKNOWN
+    };
+    SyntaxKind syntaxKind;
     /// The difference.
     std::string BodyL, BodyR;
-    SyntaxDifference() : NonFunctionDifference(SynDiff){};
-    SyntaxDifference(std::string name,
+    // Information about definitions of differing 'object'.
```

Comment:
For context instead of `differing` I used previously `last`
> I was a bit confused by the word "last" here. Maybe "differing" would be better (if it's always the case that the last object is the differing one)?

Yes, I think that the last macro should be always the differing one, but if I understand the implementation correctly the macro does not have to contain semantic difference but just syntactic one https://github.com/diffkemp/diffkemp/blob/e6cded9aefd2e51ada26c972f499fe08d442d789/diffkemp/simpll/SourceCodeUtils.cpp#L262-L263 , but we are still reporting it.